configure.ac

```diff
@@ -93,6 +93,30 @@ if test "x$squid_host_os" = "solaris" -a "x$GCC" != "x" ; then
 	AC_USE_SYSTEM_EXTENSIONS
 fi
 
+AC_MSG_CHECKING([compiler name-version])
+squid_cv_cc_name=`$CC --version | head -1 | cut -f1 -d" "`
+squid_cv_cc_majversion="unknown"
+case "$squid_cv_cc_name" in
+  gcc|clang) squid_cv_cc_majversion=`$CC -dumpversion | cut -f1 -d.` ;;
+esac
+squid_cv_cc_simplified="$squid_cv_cc_name-$squid_cv_cc_majversion"
+AC_MSG_RESULT($squid_cv_cc_simplified)
+
+# warning flags
+AS_IF([test "$squid_cv_cc_simplified" = "gcc-4"],[
+    AC_MSG_NOTICE([skipping -Wextra on $squid_cv_cc_simplified])
+],[
+    # -Werror must be first
+    SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED([-Werror],[squid_cv_cc_w_error])
+    SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED([-Wextra],[squid_cv_cc_w_extra])
+    SQUID_CC_ADD_CXXFLAG_IF_SUPPORTED([-Wno-unused-private-field],[squid_cv_cc_no_unused_private_field])
```

Comment:
It would be nice to document (in configure.ac comment) what is wrong with `-Wunused-private-field`. That is, _why_ we are disabling that warning. I do not insist on this.

Reply:
Adding